packages/auth-construct/src/construct.ts

```diff
@@ -605,14 +615,26 @@ export class AmplifyAuth
      */
     const shouldMapEmailAttributes = loginOptions.email && !loginOptions.phone;
     const result: IdentityProviderSetupResult = {
-      oauthMappings: {},
+      oAuthMappings: {},
       providersList: [],
+      oAuthSettings: undefined,
     };
     // external providers
     const external = loginOptions.externalProviders;
     if (!external) {
       return result;
     }
+    // make sure logout/callback urls are not empty
+    if (external.logoutUrls.length === 0) {
+      throw Error(
+        'You must define logoutUrls when configuring external login providers.'
+      );
+    }
+    if (external.callbackUrls.length === 0) {
+      throw Error(
+        'You must define callbackUrls when configuring external login providers.'
+      );
+    }
     if (external.google) {
```

Comment:
`if(external....){...` has a lot of repeat code as well. We can follow up to clean them up

Reply:
yeah, though a lot of these have a lot of custom mapping logic, some of are direct mappings and others are customized

Reply:
still can be cleaned up, at least, make a function for each of them